File: Audiograph/APIHistory.vb
Public Class frmAPIHistory
    Private Sub ResizeOps(sender As Object, e As EventArgs) Handles Me.Resize
        ltvStatus.Width = Me.Width - 40
        ltvStatus.Height = Me.Height - 92
    End Sub

    Private Sub ExitForm(sender As Object, e As EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

    ' updates list view ~60 times per second, i hate that i have to do this but vb sucks
    Public Sub AddListView(sender As Object, e As EventArgs) Handles tmrListUpdate.Tick
        ' compare count of list and count of listview
        If apihistory.Count > ltvStatus.Items.Count Then
            ' add every item that is not already added to the listview
            For count As Integer = ltvStatus.Items.Count To apihistory.Count - 1
                ltvStatus.Items.Insert(0, apihistory(count)(0)).SubItems.AddRange({apihistory(count)(1), apihistory(count)(2), apihistory(count)(3), apihistory(count)(4), apihistory(count)(5)})
            Next
        End If

        ' check for max items
        If ltvStatus.Items.Count > 100000 Then
            ltvStatus.Items.Clear()
        End If
    End Sub

    Private Sub ClearHistory(sender As Object, e As EventArgs) Handles btnClear.Click
        apihistory.Clear()
        ltvStatus.Items.Clear()
    End Sub
End Class

File: Audiograph/APIHistory.cs
using System;

namespace Audiograph
{
    public partial class frmAPIHistory
    {
        public frmAPIHistory()
        {
            InitializeComponent();
        }
        private void ResizeOps(object sender, EventArgs e)
        {
            ltvStatus.Width = Width - 40;
            ltvStatus.Height = Height - 92;
        }

        private void ExitForm(object sender, EventArgs e)
        {
            Close();
        }

        // updates list view ~60 times per second, i hate that i have to do this but vb sucks
        public void AddListView(object sender, EventArgs e)
        {
            // compare count of list and count of listview
            if (Utilities.apihistory.Count > ltvStatus.Items.Count)
            {
                // add every item that is not already added to the listview
                for (int count = ltvStatus.Items.Count, loopTo = Utilities.apihistory.Count - 1; count <= loopTo; count++)
                    ltvStatus.Items.Insert(0, Utilities.apihistory[count][0]).SubItems.AddRange(new[] { Utilities.apihistory[count][1], Utilities.apihistory[count][2], Utilities.apihistory[count][3], Utilities.apihistory[count][4], Utilities.apihistory[count][5] });
            }

            // check for max items
            if (ltvStatus.Items.Count > 100000)
            {
                ltvStatus.Items.Clear();
            }
        }

        private void ClearHistory(object sender, EventArgs e)
        {
            Utilities.apihistory.Clear();
            ltvStatus.Items.Clear();
        }
    }
}